python/nano/src/bigdl/nano/utils/pytorch/convert.py

```diff
@@ -42,6 +43,17 @@ def export_to_onnx(model, input_sample=None, onnx_path="model.onnx", dynamic_axe
     forward_args = get_forward_args(model)
     tensor_args = get_conditional_args(model)
     input_sample = get_input_example(model, input_sample, forward_args)
+
+    if isinstance(input_sample, Sequence) and isinstance(input_sample[-1], Dict):
```

Comment:
what if dict in input_sample is not at last?

Reply:
The logic I design the test cases is divided into two situations. 
- The **dict** type is at last
- The **dict** type is not at the last (any position except the last). 

Pytorch onnx team has a function **_decide_input_format()** to process the input sample. 
- When the **dict** type is not at the last, we don't need to do anything additional, just feed the original input _sample.
- When encountering the situation that the last item in the input sample is a **dict** type, the function will treat it as a dictionary containing optional arguments and resolve it  (read the keys of the dictionary, check if it is among the arguments of the forward function. If yes, then append it to the input list. If no, do nothing when the arg has no default value). So if we simply leave the input of **dict** type at the last, it won't be put in the input list by onnx by default, so I did a small trick here.  

For reference, the **_decide_input_format()** is here. 
https://github.com/pytorch/pytorch/blob/main/torch/onnx/utils.py#:~:text=def%20_decide_input_format(model%2C%20args)%3A

Reply:
yeah now I fully understand the logic, the `keys[0] not in forward_args` also contributes in the whole logic. Please do add some comment to this code snap :)